sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,237 @@
+# Author: Henry Lin <hlin117@gmail.com>
+
+from __future__ import division, absolute_import
+
+import numbers
+import numpy as np
+import warnings
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.preprocessing.data import _transform_selected
+from sklearn.utils.validation import (
+    check_array,
+    check_is_fitted,
+    column_or_1d,
+    FLOAT_DTYPES
+)
+
+
+class KBinsDiscretizer(BaseEstimator, TransformerMixin):
+    """Bins continuous data into k equal width intervals.
+
+    Parameters
+    ----------
+    n_bins : int or array-like, shape (n_features,) (default=2)
+        The number of bins to produce. The intervals for the bins are
+        determined by the minimum and maximum of the input data.
+        Raises ValueError if ``n_bins < 2``.
+
+        If ``n_bins`` is an array, and there is a ignored feature at
+        index ``i``, ``n_bins[i]`` will be ignored.
+
+    ignored_features : int array-like (default=None)
+        Column indices of ignored features. (Example: Categorical features.)
+        If ``None``, all features will be discretized.
+
+    Attributes
+    ----------
+    offset_ : float array, shape (n_features,)
+        Minimum value per feature in ``X``. An ignored feature at index
+        ``i`` will have ``offset_[i] == np.nan``. Used for the transformation
```

Comment:
> I think you can more explicitly say somewhere (in an attribute description or under Notes or Examples) that bin edges for feature i are defined as offset_[i] + bin_width_[i] * np.arange(n_bins_[i] + 1); or is it clearer as np.concatenate([-np.inf, offset_[i] + bin_width_[i] * np.arange(1, n_bins_[i]), np.inf])?

I'd say the latter is more correct. Some users might miss that there is a bin before the first bin edge.